manifests/params.pp

```diff
@@ -37,14 +37,7 @@
   $service_ensure = running
   $service_enable = true
   $service_manage = true
-
-  # NOTE: Workaround for the following issue:
-  # https://tickets.puppetlabs.com/browse/PUP-5296
-  if $::osfamily == 'redhat' {
-    $service_provider = 'redhat'
-  } else {
-    $service_provider = undef
-  }
```

Comment:
I think we need to keep this hack for some time because the addressed issue has been resolved only in Puppet >= 4.5.0. We can just check that `$service_provider` is `undef` before setting it to `redhat`:

```puppet
if $::osfamily == 'redhat' and $service_provider == under {
  ...
```